tfhe/src/high_level_api/global_state.rs

```diff
@@ -122,10 +122,23 @@ pub(in crate::high_level_api) fn device_of_internal_keys() -> Option<crate::Devi
     })
 }
 
+/// This returns the [tag](crate::Tag) stored in the internal server key
+#[inline]
+pub(in crate::high_level_api) fn tag_of_internal_server_key() -> crate::Result<crate::Tag> {
+    INTERNAL_KEYS.with(|keys| {
+        let cell = keys.borrow();
+        Ok(match cell.as_ref().ok_or(UninitializedServerKey)? {
+            InternalServerKey::Cpu(cpu_key) => cpu_key.tag.clone(),
+            #[cfg(feature = "gpu")]
+            InternalServerKey::Cuda(cuda_key) => cuda_key.tag.clone(),
```

Comment:
how does this work properly for GPU ? 🤔 we have two values that could be returned here ?

also what do you think about trying to use if cfg!(feature = "gpu") instead of the attributes to manage inline feature stuff ?

Reply:
nevermind did not see it was a match 😵‍💫